### PR TITLE
Integrate wiki platform source

### PR DESCRIPTION
**Pull request by peter-gy**
### Goals

- Integrate the wiki platform developed under [enermaps/doc/dev](https://github.com/enermaps/doc/tree/dev), as discussed with @gperonato
- Develop the platform with automated translation compatibility in mind
- Authenticate wiki editors via GitHub on the platform
  - For this purpose I set up a new OAuth App for the organisation, accessible under the developer settings

### Notes

- The [enermaps/doc/data](https://github.com/enermaps/doc/tree/data) branch serves currently as the remote "database" of the wiki. It is necessary to handle content synchronisation when running translations automatically

### CODE REVIEW

File: wiki/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+FROM ubuntu:18.04
```
Comment (gperonato) [inline wiki/Dockerfile]: Would it be possible to bump to 20.04 (the one used in the project)?

File: wiki/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: "3.2"
```
Comment (gperonato) [inline wiki/docker-compose.yml]: Would it be possible to bump to 3.8? And possibly move the file to the root directory (and rename it `docker-compose-wiki.yml`)?

**Review by gperonato — COMMENTED** [review]
Thank you for your PR!
I have added a few comments. They are mainly for consistency with the project guidelines/choices. But since you will be hosting the service(s) yourself, they are of course just suggestions.
Would you mind also adding a small section to the README in the main directory?

File: wiki/Dockerfile (hunk 1 above)
Comment (peter-gy) [reply wiki/Dockerfile]: Unfortunately the dependency graph of the underlying base wiki platform (`gollum`) is pretty intricate, hence `ubuntu:18.04` is the stable choice currently.

File: wiki/docker-compose.yml (hunk 2 above)
Comment (peter-gy) [reply wiki/docker-compose.yml]: Of course. I would propose to rename it to `docker-compose-doc.yml` to be consistent with the directory name (`doc` as per your request).

**Review by gperonato — COMMENTED** [review]
Hi Péter,

Thank you for the changes. The README looks good!
Just FYI, I have this error while building from the `docker-compose-doc.yml`.
```
enermaps-doc | fatal: repository '' does not exist
enermaps-doc | 2021-11-08 07:49:48,847 [app.scheduled_jobs] INFO     Refreshing AUTH_USERS
enermaps-doc | Traceback (most recent call last):
enermaps-doc |   File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
enermaps-doc |     "__main__", mod_spec)
enermaps-doc |   File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
enermaps-doc |     exec(code, run_globals)
enermaps-doc |   File "/root/docker-run/app/run.py", line 19, in <module>
enermaps-doc |     refresh_auth_users()
enermaps-doc |   File "/root/docker-run/app/scheduled_jobs.py", line 28, in refresh_auth_users
enermaps-doc |     auth_users = get_authenticated_user_string()
enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 52, in get_authenticated_user_string
enermaps-doc |     gh: Github = __git_hub_instance()
enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 29, in __git_hub_instance
enermaps-doc |     token = __extract_personal_access_token_from_repo_url(repo_url=wiki_remote_url)
enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 22, in __extract_personal_access_token_from_repo_url
enermaps-doc |     return re.findall(r'https:\/\/(.*)@', repo_url)[0]
enermaps-doc | IndexError: list index out of range
```

Also, is the `docker-compose.yml` in the `/doc` directory still used?

**Comment by peter-gy** [comment]
> Hi Péter,
> 
> Thank you for the changes. The README looks good! Just FYI, I have this error while building from the `docker-compose-doc.yml`.
> 
> ```
> enermaps-doc | fatal: repository '' does not exist
> enermaps-doc | 2021-11-08 07:49:48,847 [app.scheduled_jobs] INFO     Refreshing AUTH_USERS
> enermaps-doc | Traceback (most recent call last):
> enermaps-doc |   File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
> enermaps-doc |     "__main__", mod_spec)
> enermaps-doc |   File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
> enermaps-doc |     exec(code, run_globals)
> enermaps-doc |   File "/root/docker-run/app/run.py", line 19, in <module>
> enermaps-doc |     refresh_auth_users()
> enermaps-doc |   File "/root/docker-run/app/scheduled_jobs.py", line 28, in refresh_auth_users
> enermaps-doc |     auth_users = get_authenticated_user_string()
> enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 52, in get_authenticated_user_string
> enermaps-doc |     gh: Github = __git_hub_instance()
> enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 29, in __git_hub_instance
> enermaps-doc |     token = __extract_personal_access_token_from_repo_url(repo_url=wiki_remote_url)
> enermaps-doc |   File "/root/docker-run/app/editor_auth_handler.py", line 22, in __extract_personal_access_token_from_repo_url
> enermaps-doc |     return re.findall(r'https:\/\/(.*)@', repo_url)[0]
> enermaps-doc | IndexError: list index out of range
> ```
> 
> Also, is the `docker-compose.yml` in the `/doc` directory still used?

Hi Giuseppe!

The cause of this error when you run the built container is that you have probably not specified the required environment variables properly. In my latest commit (ad5c617) I adjusted the startup script so that it fails with a meaningful error message in the case of missing env vars.

Yes, `./doc/docker-compose.yml` still can be used. It is going to look for `./doc/.env` to load the environment variables.

Simlarly, `./docker-compose-doc.yml` can also be used, and it is going to look for `./.env-doc` to load the required env values.

I attached the actual values to be used to you in email, encrypted. Putting those values into the `.env` files will let you to run the container locally without errors.

**Review by gperonato — APPROVED** [review]

**Pull request merged**